src/commonTest/kotlin/fr/acinq/lightning/tests/utils/LightningTestSuite.kt

```diff
@@ -10,13 +10,11 @@ abstract class LightningTestSuite {
         setLogger()
     }
 
-    @ThreadLocal
     companion object {
-        private var loggerIsSet = false
         fun setLogger() {
-            if (loggerIsSet) return
-            setLightningLoggerFactory(testLightningLoggerFactory)
-            loggerIsSet = true
+            if (!isLightningLoggerFactorySet()) {
```

Comment:
This looks suspicious: we he had one "guard" (`loggerIsSet`) per thread, now we have one global guard `LightningLoggerFactory` with no locks/guards to synchronize access to it from different threads ?

Reply:
Right. How about putting that code inside an `init` block if the companion object? Should be enough right? Since it's a singleton.

Reply:
Done in [d3568d1](https://github.com/ACINQ/lightning-kmp/pull/348/commits/d3568d152381bc3444a08510496ec37cef1ace26).

Reply:
This was handled in #359.